libs/widget-lib/src/IframeCowEventEmitter.ts

```diff
@@ -1,18 +1,26 @@
 import { SimpleCowEventEmitter, CowEventListener, CowEventListeners, CowEvents } from '@cowprotocol/events'
 import { WidgetMethodsEmit } from './types'
-import { listenToMessageFromWindow } from './messages'
+import { WindowListener, listenToMessageFromWindow, stopListeningWindowListener } from './messages'
 
+let i = 1
 export class IframeCowEventEmitter {
   private eventEmitter: SimpleCowEventEmitter = new SimpleCowEventEmitter()
+  private listeners: CowEventListeners = []
+  private widgetListener: WindowListener
+  private instanceNumber = i++
 
-  constructor(contentWindow: Window, private listeners: CowEventListeners = []) {
-    // Subscribe to events
+  constructor(private contentWindow: Window, listeners: CowEventListeners = []) {
```

Comment:
Is `listeners` param no longer private? Or every parameter after it's used becomes private as well?

Reply:
it's still private, but actually adding `private` there introduces a non obvious bug. 

Basically, overrides the value of the local instance prop with the param value, and this will make that the de-registering of the events will de-register the new events and not the old ones (this de-registering will not have effect, but it will register the event twice).